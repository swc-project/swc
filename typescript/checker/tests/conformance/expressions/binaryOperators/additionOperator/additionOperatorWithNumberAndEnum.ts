enum E { a, b }
enum F { c, d }

var a: number;
var b: E;
var c: E | F;

var r1 = a + a;
var r2 = a + b;
var r3 = b + a;
var r4 = b + b;

var r5 = 0 + a;
var r6 = E.a + 0;
var r7 = E.a + E.b;
var r8 = E['a'] + E['b'];
var r9 = E['a'] + F['c'];

var r10 = a + c;
var r11 = c + a;
var r12 = b + c;
var r13 = c + b;
var r14 = c + c;
