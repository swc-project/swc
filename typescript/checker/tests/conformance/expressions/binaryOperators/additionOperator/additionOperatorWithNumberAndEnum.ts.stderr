error: ShouldIncludeUndefinedType {
    span: Span {
        lo: BytePos(
            37,
        ),
        hi: BytePos(
            38,
        ),
        ctxt: #0,
    },
}
 --> $DIR/tests/conformance/expressions/binaryOperators/additionOperator/additionOperatorWithNumberAndEnum.ts:4:5
  |
4 | var a: number;
  |     ^

error: ShouldIncludeUndefinedType {
    span: Span {
        lo: BytePos(
            52,
        ),
        hi: BytePos(
            53,
        ),
        ctxt: #0,
    },
}
 --> $DIR/tests/conformance/expressions/binaryOperators/additionOperator/additionOperatorWithNumberAndEnum.ts:5:5
  |
5 | var b: E;
  |     ^

error: ShouldIncludeUndefinedType {
    span: Span {
        lo: BytePos(
            62,
        ),
        hi: BytePos(
            63,
        ),
        ctxt: #0,
    },
}
 --> $DIR/tests/conformance/expressions/binaryOperators/additionOperator/additionOperatorWithNumberAndEnum.ts:6:5
  |
6 | var c: E | F;
  |     ^

